### PR TITLE
Bug Fix: chat completions API calls need `model_id`

### PR DESCRIPTION
**Pull request by tybrs**
## Description

Users must pass model parameter explicitly to use /v1/chat/completions API call:

```
curl http://localhost:8088/v1/chat/completions \
    -X POST \
    -d '{"model": "meta-llama/Meta-Llama-Guard-2-8B", "messages": [{"role": "user", "content": "Say this is a test!"}]}' \
    -H 'Content-Type: application/json'
```

Using `ChatHuggingFace` in langchain this looks like the following:
```
llm_guard_chat = ChatHuggingFace(llm=llm_guard, model_id=safety_guard_model)
llm_guard_chat.invoke([{"role": "user", "content": input.text}]).content
```
If you do not add a `model_id` it will use the `_resolve_model_id` method to get the `model_id` but this does not work for locally deployed TGI services (because it uses huggingface_hub `list_inference_endpoints` function which only lists hugging face endpoints for account). This issue can be found here  https://github.com/langchain-ai/langchain/issues/17779. Current implementation errors as follows:

```
llm_engine_hf= ChatHuggingFace(llm=llm_guard)
^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/langchain_community/chat_models/huggingface.py", line 54, in init
self._resolve_model_id()
File "/usr/local/lib/python3.11/site-packages/langchain_community/chat_models/huggingface.py", line 158, in _resolve_model_id
raise ValueError(
ValueError: Failed to resolve model_id Could not find model id for inference server provided: http://xx.xx.xx.xxx/
Make sure that your Hugging Face token has access to the endpoint.
```

This PR adds a workaround to this issue by adding a `DEFAULT_MODEL`  variable and a way to try to get model id from `/info` endpoint. I also moved the `ChatHuggingFace` to run at start. This should improve latency of API calls by 1-2 seconds.

## Issues

 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

`langchain-community`

## Tests

1. I rain two local TGI services of both LlamaGuard1 and 

```
docker run -p 8087:80 -v $PWD/data:/data  -e HTTPS_PROXY=$https_proxy -e HTTP_PROXY=$https_proxy ghcr.io/huggingface/text-generation-inference --model-id meta-llama/LlamaGuard-7b
```

```
docker run -p 8087:80 -v $PWD/data:/data  -e HTTPS_PROXY=$https_proxy -e HTTP_PROXY=$https_proxy ghcr.io/huggingface/text-generation-inference --model-id meta-llama/Meta-Llama-Guard-2-8B
```

2. Then I build and ran guardrails-tgi-server

```
docker build -t opea/gen-ai-comps:guardrails-tgi-server --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/guardrails/langchain/docker/Dockerfile .
```

```
docker run -p 9090:9090  -e https_proxy=$https_proxy -e http_proxy=$http_proxy -e no_proxy=$no_proxy -e HUGGINGFACEHUB_API_TOKEN=$HUGGINGFACEHUB_API_TOKEN -e SAFETY_GUARD_ENDPOINT="http://localhost:8088" --network="host" opea/gen-ai-comps:guardrails-tgi-server
```

3. Tested with following curl:
```
$ curl http://localhost:9090/v1/guardrails -X POST   -d '{"text": "i am going to kill you"}'   -H 'Content-Type: application/json'
{"id":"e6a511430e2d5158d2923a4099502945","text":"Violated policies: Violent Crimes, please check your input."}twilbers@mlp-prod-skx-5675:/localdisk/twilbers/docker$
```

### CODE REVIEW

**Review by lvliang-intel — APPROVED** [review]

**Comment by tybrs** [comment]
@lvliang-intel  Thanks for the review. Should we assign another person for review?

File: comps/guardrails/langchain/guardrails_tgi_gaudi.py [hunk 1]
```diff
@@ -38,6 +41,18 @@ def get_unsafe_dict(model_id="meta-llama/LlamaGuard-7b"):
         }
 
 
+def get_tgi_service_model_id(endpoint_url, default=DEFAULT_MODEL):
+    """Returns Hugging Face repoo id for deployed service's info endpoint
```
Comment (dcmiddle) [inline comps/guardrails/langchain/guardrails_tgi_gaudi.py:45]: nit typo 
```suggestion
    """Returns Hugging Face repo id for deployed service's info endpoint
```

File: comps/guardrails/langchain/guardrails_tgi_gaudi.py [hunk 2]
```diff
@@ -66,6 +79,7 @@ def safety_guard(input: TextDoc) -> TextDoc:
 
 if __name__ == "__main__":
     safety_guard_endpoint = os.getenv("SAFETY_GUARD_ENDPOINT", "http://localhost:8080")
+    safety_guard_model = os.getenv("SAFETY_GUARD_MODEL_ID", get_tgi_service_model_id(safety_guard_endpoint))
```
Comment (dcmiddle) [inline comps/guardrails/langchain/guardrails_tgi_gaudi.py:82]: Should the instructions be updated adding this environment variable?
https://github.com/opea-project/GenAIComps/tree/main/comps/guardrails#21-setup-environment-variables

File: comps/guardrails/langchain/guardrails_tgi_gaudi.py (hunk 1 above)
Comment (tybrs) [reply comps/guardrails/langchain/guardrails_tgi_gaudi.py]: Fixed. Thanks @dcmiddle

File: comps/guardrails/langchain/guardrails_tgi_gaudi.py (hunk 2 above)
Comment (tybrs) [reply comps/guardrails/langchain/guardrails_tgi_gaudi.py:82]: Thanks!

I added the following instructions:

https://github.com/opea-project/GenAIComps/blob/9ccabd113eeb973e4b817d7c22b32baf3f79cd2a/comps/guardrails/README.md?plain=1#L55-L61

**Comment by tybrs** [comment]
@dcmiddle and @lvliang-intel  Comments addressed and conflict fixed. Thanks!

**Review by dcmiddle — APPROVED** [review]

**Review by zehao-intel — APPROVED** [review]

**Pull request merged**